### PR TITLE
Refactor documentation for Apple targets

### LINKED ISSUE #124215: Document minimum supported versions for apple operating systems to platform documentation
In https://github.com/rust-lang/rust/pull/124089#pullrequestreview-2013141782, it was pointed out that there's not an easy way to know the minimum version of an apple OS that is supported. One must kind of grep for `*_DEPLOYMENT_TARGET` and then parse the output to retrieve an answer and get an answer.

If I read the source correctly, the minimum supported versions currently are:
* [watchOS 5.0](https://github.com/rust-lang/rust/blob/54692c3d0b873a5dc055f2f35081d0d6a2410a49/compiler/rustc_target/src/spec/base/apple/mod.rs#L369)
* [tvOS 10.0](https://github.com/rust-lang/rust/blob/54692c3d0b873a5dc055f2f35081d0d6a2410a49/compiler/rustc_target/src/spec/base/apple/mod.rs#L354C4-L354C76) 
* [visionOS 1.0](https://github.com/rust-lang/rust/blob/54692c3d0b873a5dc055f2f35081d0d6a2410a49/compiler/rustc_target/src/spec/base/apple/mod.rs#L384)
* [macOS 10.12 for non arm and 11.0 for arm macOS](https://github.com/rust-lang/rust/blob/54692c3d0b873a5dc055f2f35081d0d6a2410a49/compiler/rustc_target/src/spec/base/apple/mod.rs#L264-L276)
* [iOS 14 for arm64e, 13.1 for mac catalyst and 10.0 otherwise](https://github.com/rust-lang/rust/blob/54692c3d0b873a5dc055f2f35081d0d6a2410a49/compiler/rustc_target/src/spec/base/apple/mod.rs#L320-L329)

The versions are occasionally bumped up as in https://github.com/rust-lang/compiler-team/issues/556 and so it would be good write them down in a more human readable form.

This has lead to various [code comments about version support for various platforms being a little inconsistent](https://github.com/rust-lang/rust/blob/453ceafce32ef8108c604bca5e165ab41d3d6d8c/library/std/src/sys/pal/unix/fd.rs#L248-L251). It is possible this comment is to mean "minimum version of this syscall" but that is unclear.

I will admit that I'm not certain as to the best way to do this. Adding these to [the platform support docs](https://doc.rust-lang.org/rustc/platform-support.html) seems like the right choice. Perhaps also in [the specific platform support page](https://doc.rust-lang.org/rustc/platform-support/apple-tvos.html)

Some of the tier 3 pages list a minimum and some do not. [`arm64e-apple-ios` for example lists 12.0+](https://doc.rust-lang.org/rustc/platform-support/arm64e-apple-ios.html) and [`arm64e-apple-darwin` lists 11+](https://doc.rust-lang.org/rustc/platform-support/arm64e-apple-darwin.html) however [watchOS](https://doc.rust-lang.org/rustc/platform-support/apple-watchos.html) and [tvOS](https://doc.rust-lang.org/rustc/platform-support/apple-tvos.html) do not.

Perhaps related: I am unfamiliar with target triples `arm64e-apple-ios` and `arm64e-apple-darwin`. Are these actually used?

CC: @madsmtm

### PR DESCRIPTION
**Pull request by madsmtm**
Refactor the documentation for Apple targets in `rustc`'s platform support page to make it clear what the supported OS version is and which environment variables are being read (`*_DEPLOYMENT_TARGET` and `SDKROOT`). This fixes https://github.com/rust-lang/rust/issues/124215.

Note that I've expanded the `aarch64-apple-ios-sim` maintainers @badboy and @deg4uss3r to include being maintainer of all `*-apple-ios-*` targets. If you do not wish to be so, please state that, then I'll explicitly note that in the docs.

Additionally, I've added myself as co-maintainer of most of these targets.

r? @thomcc

I think the documentation you've previously written on tvOS is great, have mostly modified it to have a more consistent formatting with the rest of the Apple target.

I recognize that there's quite a few changes here, feel free to ask about any of them!

---

CC @simlay @Nilstrieb

@rustbot label O-apple

### CODE REVIEW

**Comment by badboy** [comment]
works for me. The targets are similar enough and I use them that I can try to help if things break.

**Review by simlay — APPROVED** [review]

**Comment by madsmtm** [comment]
r?

**Comment by oli-obk** [comment]
I don't have an opinion on the changes, but they look like an improvement and y'all seem happy with it.

@bors r+ rollup

File: src/doc/rustc/src/SUMMARY.md [hunk 1]
```diff
@@ -16,13 +16,17 @@
...
-    - [arm64e-apple-darwin.md](platform-support/arm64e-apple-darwin.md)
-    - [aarch64-apple-ios-sim](platform-support/aarch64-apple-ios-sim.md)
     - [arm64ec-pc-windows-msvc](platform-support/arm64ec-pc-windows-msvc.md)
+    - [\*-apple-darwin](platform-support/apple-darwin.md)
+        - [i686-apple-darwin](platform-support/i686-apple-darwin.md)
+        - [x86_64h-apple-darwin](platform-support/x86_64h-apple-darwin.md)
+        - [arm64e-apple-darwin.md](platform-support/arm64e-apple-darwin.md)
+    - [\*-apple-ios](platform-support/apple-ios.md)
+        - [\*-apple-ios-macabi](platform-support/apple-ios-macabi.md)
+        - [arm64e-apple-ios.md](platform-support/arm64e-apple-ios.md)
```
Comment (weihanglo) [inline src/doc/rustc/src/SUMMARY.md:26]: the `.md` suffix is probably not needed?

```suggestion
        - [arm64e-apple-darwin](platform-support/arm64e-apple-darwin.md)
    - [\*-apple-ios](platform-support/apple-ios.md)
        - [\*-apple-ios-macabi](platform-support/apple-ios-macabi.md)
        - [arm64e-apple-ios](platform-support/arm64e-apple-ios.md)
```

**Pull request merged**